core/src/test/kotlin/io/specmatic/core/RunContractTestsUsingScenario.kt

```diff
@@ -768,9 +765,15 @@ paths:
                     val result = securitySchema.matches(request, Resolver())
                     assertThat(result).isInstanceOf(Result.Success::class.java)
 
-                    val actualValue = extractValue(request)
-                    val expectedValue = extractValue(request)
-                    assertThat(actualValue).isEqualTo(expectedValue)
+                    val schemesToCheck = when(securitySchema) {
+                        is CompositeSecurityScheme -> securitySchema.schemes
+                        else -> listOf(securitySchema)
+                    }
+                    assertThat(schemesToCheck).allSatisfy {
+                        val actualValue = extractValue(request)
+                        val expectedValue = extractValue(request)
+                        assertThat(actualValue).isEqualTo(expectedValue)
```

Comment:
both the above seem to extract the same value, so won't the assertion just pass?